juju/client/jujudata.py

```diff
@@ -75,7 +75,10 @@ def refresh(self):
 
     def current_controller(self):
         '''Return the current controller name'''
-        return self._load_yaml('controllers.yaml', 'current-controller')
+        try:
+            return self._load_yaml('controllers.yaml', 'current-controller')
+        except FileNotFoundError:
+            return None
```

Comment:
This is the spot at which we should raise the `JujuConnectionError` ☝️

Reply:
Fixed in https://github.com/juju/python-libjuju/pull/937/commits/747dc771710bd0295c4480c0927d07a144e39b17